### PR TITLE
Migrate "runtime-metrics" test from groovy to java

### PR DESCRIPTION
**Pull request by breedx-splk**
Related to #7195

### CODE REVIEW

File: instrumentation/runtime-telemetry/runtime-telemetry-java17/javaagent/src/test/java/io/opentelemetry/instrumentation/javaagent/runtimemetrics/java17/JmxRuntimeMetricsTest.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+ */
+
+package io.opentelemetry.instrumentation.javaagent.runtimemetrics.java17;
+
+import io.opentelemetry.instrumentation.testing.junit.AgentInstrumentationExtension;
+import io.opentelemetry.instrumentation.testing.junit.InstrumentationExtension;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+public class JmxRuntimeMetricsTest {
```
Comment (jeanbisutti) [inline instrumentation/runtime-telemetry/runtime-telemetry-java17/javaagent/src/test/java/io/opentelemetry/instrumentation/javaagent/runtimemetrics/java17/JmxRuntimeMetricsTest.java:13]:
```suggestion
class JmxRuntimeMetricsTest {
```

**Review by jeanbisutti — APPROVED** [review]

**Comment by jeanbisutti** [comment]
Great! It could help to test JVM metrics with a GraalVM native executable.

File: testing-common/src/main/java/io/opentelemetry/instrumentation/testing/junit/InstrumentationExtension.java [hunk 2]
```diff
@@ -76,6 +80,13 @@ public List<MetricData> metrics() {
     return testRunner.getExportedMetrics();
   }
 
+  /** Returns a list of metrics, filtered by instrumentation name */
+  public List<MetricData> instrumentationMetrics(String instrumentationName) {
```
Comment (mateuszrzeszutek) [inline testing-common/src/main/java/io/opentelemetry/instrumentation/testing/junit/InstrumentationExtension.java:84]: We don't have analogous methods for any other signal; and it's only used in this class. WDYT about making it private?
```suggestion
  private List<MetricData> instrumentationMetrics(String instrumentationName) {
```

Comment (breedx-splk) [reply testing-common/src/main/java/io/opentelemetry/instrumentation/testing/junit/InstrumentationExtension.java:84]: I intended it to be a public helper, analogous to `metrics()` but filtered to a specific instrumentation. It's only used internally now, but I could see it being useful elsewhere. If you think it's too much YAGNI then I will make it private.

Comment (mateuszrzeszutek) [reply testing-common/src/main/java/io/opentelemetry/instrumentation/testing/junit/InstrumentationExtension.java:84]: Let's make it private -- there's just one use case for that now; and I think all 3 signals should have very similar API, so if we're introducing some new thing here we should do similar thing everywhere.

Comment (breedx-splk) [reply testing-common/src/main/java/io/opentelemetry/instrumentation/testing/junit/InstrumentationExtension.java:84]: Ok I relent.  :) I think you have a higher standard for test code than I do.

**Review by mateuszrzeszutek — APPROVED** [review]

File: testing-common/src/main/java/io/opentelemetry/instrumentation/testing/junit/InstrumentationExtension.java (hunk 2 above)
Comment (mateuszrzeszutek) [reply testing-common/src/main/java/io/opentelemetry/instrumentation/testing/junit/InstrumentationExtension.java]: Hah, I'm just thinking that YAGNI (probably) when it comes to making that method public

**Pull request merged**